### ISSUE TITLE
Standard L2 Genesis: Semgrep Rule to ensure all predeploys are non-initializable

### ISSUE THREAD

**Issue by maurelian** [ethereum-optimism/optimism#12580]
https://github.com/ethereum-optimism/design-docs/pull/97/files#r1790854263

**Comment by maurelian** [comment]
@AmadiMichael, I think you're going to take this on as part of https://github.com/ethereum-optimism/design-docs/pull/144. 

Do you want to assign yourself this issue or do you have another one already?

**Comment by AmadiMichael** [comment]
Yeah, I'll assign this to myself though there's a more general issue here https://github.com/ethereum-optimism/optimism/issues/12634
